### PR TITLE
Various fixes in prep for 0.8 release

### PR DESCRIPTION
**Pull request by kentonv**

### CODE REVIEW

File: c++/src/kj/common.h [hunk 1]
```diff
@@ -1387,8 +1387,34 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
...
   inline constexpr ArrayPtr(T* begin, T* end): ptr(begin), size_(end - begin) {}
+
+#if __GNUC__ && !__clang__ && __GNUC__ >= 9
+// GCC 9 added a warning when we take an initializer_list as a constructor parameter and save a
+// pointer to its content in a class member. GCC apparently imagines we're going to do something
+// dumb like this:
+//     ArrayPtr<const int> ptr = { 1, 2, 3 };
+//     foo(ptr[1]); // undefined behavior!
+// Any KJ programmer should be able to recognize that this is UB, because an ArrayPtr does not own
+// its content. That's not what this constructor is for, tohugh. This constructor is meant to allow
```
Comment (Sarcasm) [inline c++/src/kj/common.h:1398]: typo: tohugh -> though

**Comment by Sarcasm** [comment]
I see that you are fixing GCC warnings, and have issue with testing GCC 9.

I grabbed your commits until now:

```
commit d63dd055b156b4eb5ddb0ac86b3aac98f91e0563 (HEAD -> master, origin/release-prep)
Author: Kenton Varda <kenton@cloudflare.com>
Date:   Thu Apr 9 13:59:27 2020 -0500

    Yet another bogus GCC 9 warning.
    
    I wish Debian Buster had a GCC 9 package I could install to iterate on this faster, rather than waiting for Cygwin CI.
```

And I have these 2 warnings left (but I'm on Linux, not Cygwin):

```
[79/168] Building CXX object c++/src/capnp/CMakeFiles/capnp.dir/schema.c++.o
../c++/src/capnp/schema.c++: In member function ‘capnp::Type capnp::Schema::BrandArgumentList::operator[](capnp::uint) const’:
../c++/src/capnp/schema.c++:426:12: warning: ‘*((void*)& result +4)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  426 |     result = Type(static_cast<schema::Type::Which>(binding.which), binding.schema);
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[105/168] Building CXX object c++/src/capnp/CMakeFiles/capnpc.dir/compiler/node-translator.c++.o
../c++/src/capnp/compiler/node-translator.c++: In member function ‘virtual capnp::uint capnp::compiler::NodeTranslator::StructLayout::Group::addData(capnp::uint)’:
../c++/src/capnp/compiler/node-translator.c++:491:45: warning: ‘*((void*)& bestLocation +4)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  491 |         return parentDataLocationUsage[*best].allocateFromHole(
      |                                             ^
../c++/src/capnp/compiler/node-translator.c++: In member function ‘capnp::Orphan<capnp::List<capnp::schema::Annotation, capnp::Kind::STRUCT> > capnp::compiler::NodeTranslator::compileAnnotationApplications(capnp::List<capnp::compiler::Declaration::AnnotationApplication, capnp::Kind::STRUCT>::Reader, kj::StringPtr)’:
../c++/src/capnp/compiler/node-translator.c++:3280:14: warning: ‘*((void*)& kind +2)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 3280 |       } else if (*kind != Declaration::ANNOTATION) {
      |              ^~
```

**Comment by kentonv** [comment]
@Sarcasm Our test script actually disables `-Wmaybe-uninitialized` under GCC (but keeps it when using Clang) because GCC emits several such warnings that seem clearly wrong, which is what you're seeing. We also run the whole test suite under `valgrind` which should catch any actual uninitialized variable usage.

**Comment by kentonv** [comment]
Whelp, Cygwin is back to hanging for an hour at the end of tests, like it used to before updating to GCC 9. Cool.

**Pull request merged**